### PR TITLE
Add CloudWatch Log Metric Filter resource

### PR DESCRIPTION
**Pull request by ajkerrigan**
- Add the `aws.log-metric-filter` resource
- Include an `alarm` filter to match existing alarms based on metric name and namespace

Note: The overloading/repetition of the terms "metric" and "filter" in this resource name seem to invite confusion. First, class names ending with Filter tend to be... filters. And second, we already have a widely used metrics filter. I've tried to name the new classes explicitly, but the cost is some pretty awkward names. Better/alternative ideas welcome!

Addresses #6601

### CODE REVIEW

File: c7n/resources/cw.py [hunk 1]
```diff
@@ -466,6 +468,87 @@ def get_dimensions(self, resource):
...
+            for r in resources:
+                r[self.annotation_key] = list(itertools.chain(*(
+                    self.manager.retry(
+                        client.describe_alarms_for_metric,
+                        Namespace=t['metricNamespace'],
+                        MetricName=t['metricName'])['MetricAlarms']
+                    for t in r.get('metricTransformations', ())
+                )))
+        else:
+            alarms = Alarm(self.manager.ctx, {}).resources()
```
Comment (kapilt) [inline c7n/resources/cw.py:522]: alarms = self.get_resource_manager('aws.alarm') is the preferred mechanism for getting another resource manager

File: c7n/resources/cw.py [hunk 2]
```diff
@@ -466,6 +468,87 @@ def get_dimensions(self, resource):
         return [{'Name': 'LogGroupName', 'Value': resource['logGroupName']}]
 
 
+@resources.register('log-metric-filter')
```
Comment (kapilt) [inline c7n/resources/cw.py:471]: can we drop the -filter part of this in the resource name, feels a little confusing talking about the filter's filters. and in the end  it is just a metric from logs.

**Review by kapilt — COMMENTED** [review]
lgtm, thanks. however there are bunch of commits from other merged here, which are in the diff.

**Comment by ajkerrigan** [comment]
> lgtm, thanks. however there are bunch of commits from other merged here, which are in the diff.

good call, think i botched a local merge before my last push. looking cleaner now, thanks.

**Review by JohnHillegass — APPROVED** [review]
LGTM! Approving now based on recent cleanup.

**Review by kapilt — APPROVED** [review]
thanks, lgtm

**Pull request merged**